### PR TITLE
Add required Github secret. Closes #15. Breaking change, bump to 1.1.0

### LINKED ISSUE #15: add secret token
It would be awesome to be able to place a secret token in stage-ci to take advantage of the webhook secret token functionality that [github offers](https://developer.github.com/webhooks/securing/).

### PR DESCRIPTION
**Pull request by jimthedev**
This adds a github webhook secret, required checks for env key, bumps to 2.0.0 due to breaking changes. Should be merged in tandem with docs PR at https://github.com/zpnk/stage.now/issues/13

### CODE REVIEW

**Comment by jimthedev** [comment]
@zpnk Ok this is all set. Docs PR is done so both should be able to go whenever you are done reviewing.

File: src/server.js [hunk 1]
```diff
@@ -17,6 +17,18 @@ server.get('/', (request, response) => {
 });
 
 server.post('/', (request, response) => {
+  // Don't log but give a very specific error. We don't want to fill the logs.
+  if (!isGithubRequestCrypographicallySafe(request, process.env.GITHUB_WEBHOOK_SECRET)) {
```
Comment (amccloud) [inline src/server.js]: This stuff looks great. Really appreciate you jumping in and helping us with this project. I have a few changes I'd recommend that should be made:

- Can you move this logic into the `github` function? The `success` param can become a `status` number to support a 500. The idea is that in the future we can swap out that function with `gitlab` or `bitbucket` with the only argument being `request.body` (which is exactly what `isGithubRequestCrypographicallySafe` needs).
- I also think this should be an assertion that throws. If anyone is sending malformed / request to this endpoint I would want my monitoring to alert me of the exception.

**Review by amccloud — DISMISSED** [review]

**Comment by jimthedev** [comment]
Good idea. I'll try moving it inside.

File: src/server.js (hunk 1 above)
Comment (jimthedev) [reply src/server.js]: Good idea. I'll try moving it inside.

**Comment by jimthedev** [comment]
@amccloud Changes implemented and available for review. Webhook errors can be generated on a per provider basis but are thrown by the individual provider implementations. Currently only github is provided but it would be trivial to add Gitlab or others.

**Comment by zpnk** [comment]
👋  @jimthedev I really like this! I'm 👍 on merging, except for one thing: does this merit a 2.0 version bump? I understand that it's technically "breaking" because of being required, but feels kinda weird to call this v2. I guess I'm not sure how we should version, because stage-ci is consumed a bit differently than a library/framework. Do you have any thoughts on this? I think I'd be for going with v1.1.

Sorry to be a bit pedantic!

**Comment by jimthedev** [comment]
@zpnk No worries. On my repos I am pretty liberal with my version bumps because otherwise I feel like semantic versioning starts to lose meaning. For example, think of how many version of Chrome there are.

With that said I am happy to use whichever scheme you prefer. 👍

**Comment by jimthedev** [comment]
@zpnk Rebased to 1.1.0. Ready to go.

**Comment by zpnk** [comment]
🙌  Thanks for this feature @jimthedev! Great work 👍

**Comment by jimthedev** [comment]
🎉  Thanks for the merge. btw I am the author of commitizen so ChangeLine sounds interesting!

**Pull request merged**